src/main/java/org/jabref/gui/texparser/ParseTexDialogViewModel.java

```
@@ -0,0 +1,152 @@
+        return browseButton;
+    }
+
+    protected BooleanProperty searchButtonProperty() {
+        return searchButton;
+    }
+
+    protected ObservableList<Path> getSearchPathList() {
+        return searchPathList;
+    }
+
+    protected ObservableList<Path> getSelectedFileList() {
+        return selectedFileList;
+    }
+
+    protected BooleanProperty selectAllButtonProperty() {
+        return selectAllButton;
+    }
+
+    protected BooleanProperty unselectAllButtonProperty() {
+        return unselectAllButton;
+    }
+
+    protected BooleanProperty parseButtonProperty() {
+        return parseButton;
+    }
+
+    protected void browseButtonClicked() {
+        DirectoryDialogConfiguration directoryDialogConfiguration = new DirectoryDialogConfiguration.Builder()
+                .withInitialDirectory(initialPath).build();
+
+        dialogService.showDirectorySelectionDialog(directoryDialogConfiguration).ifPresent(selectedDirectory -> {
+            texDirectory.set(selectedDirectory.toAbsolutePath().toString());
+            preferencesService.setWorkingDir(selectedDirectory.toAbsolutePath());
+        });
+    }
+
+    protected void searchButtonClicked() {
+        BackgroundTask walkFileTreeTask = new WalkFileTreeView(getSearchDirectory())
+                .onRunning(() -> {
+                    browseButton.set(true);
+                    searchButton.set(true);
+                    selectAllButton.set(true);
+                    unselectAllButton.set(true);
+                    parseButton.set(true);
+                })
+                .onFinished(() -> {
+                    browseButton.set(false);
+                    searchButton.set(false);
+                })
+                .onSuccess(paths -> {
+                    searchPathList.setAll(paths);
+                    selectAllButton.set(false);
+                    unselectAllButton.set(false);
+                    parseButton.set(false);
+                });
```

Also add the onFailure method for the background task as otherwise a possible exception might be swallowed. I recently encountered that with some file download task in the linked files editor